### ISSUE TITLE
Have a github action to compile and publish a presentation on github pages

### ISSUE THREAD

**Issue by panglesd** [panglesd/slipshow#51]
Taking inspiration from https://github.com/ralexander-phi/marp-to-pages.
Thanks 8organicbits for the idea!

**Comment by panglesd** [comment]
Yes, you're right.

I think that GH actions are not for writing your presentation -- you need a "live-previewing" to make the experience nice.
So a GH action would only be useful as a way to host your presentation. I think there are several ways to do this:
- Have a section in the docs with a copy-pastable bit of yaml, to copy in a workflow file, that installs slipshow, compiles the presentation, and commits the result in the gh-pages
- Release a github action in the marketplace that does this, and have a section in the docs with a copy-pastable, much smaller, bit of yaml that uses it.

**Comment by lukstafi** [comment]
I think the most value-added part of this issue is to have a way to distribute a slipshow binary for use from GitHub actions. `opam install slipshow` takes a lot of time as it needs to compile from sources the whole dependency cone.

**Comment by panglesd** [comment]
> I think the most value-added part of this issue is to have a way to distribute a slipshow binary for use from GitHub actions.

In another project I used docker (just like in the example linked above), that's not ideal (the docker image needs to be maintained).
Another way is to download the latest github release of slipshow (it contains precompiled binaries), and use that.
Or, as you said use opam, but that uses a lot of compute power... (and I'm not sure how caching would work, wrt new slipshow versions?)

> Edited: the difference between the options you mention is rather small, and orthogonal to whether slipshow is installed from a binary or from sources with transitively from-source installed dependencies.

Yes.

**Comment by lukstafi** [comment]
> Another way is to download the latest github release of slipshow (it contains precompiled binaries), and use that.

That sounds best in terms of "doing the right thing" except I don't know how easy it is to write the yaml.

**Comment by panglesd** [comment]
Thanks a lot!! I'll add that to the docs/a github action, at some point, if you allow me!

**Comment by lukstafi** [comment]
Of course, I'd be honored. Do note that the Deploy section uses the old-school gh-pages branch approach, I don't remember what's the current recommended approach.
